spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkScan.java

```diff
@@ -194,10 +205,40 @@ protected Statistics estimateStatistics(Snapshot snapshot) {
     Map<NamedReference, ColumnStatistics> colStatsMap = Collections.emptyMap();
     if (readConf.reportColumnStats() && cboEnabled) {
       colStatsMap = Maps.newHashMap();
+      Map<Integer, Long> ndvs = Maps.newHashMap();
+      Map<Integer, Long> nullCounts = Maps.newHashMap();
+      Map<Integer, Object> minValues = Maps.newHashMap();
+      Map<Integer, Object> maxValues = Maps.newHashMap();
       List<StatisticsFile> files = table.statisticsFiles();
       if (!files.isEmpty()) {
         List<BlobMetadata> metadataList = (files.get(0)).blobMetadata();
 
+        if (readConf.deriveStatsFromManifestSessionConf()
+            || readConf.deriveStatsFromManifestTableProperty()) {
+          Map<String, Map<Integer, Long>> distinctDataFilesNullCount = Maps.newHashMap();
```

Comment:
what does "distinct" in this context mean?

Reply:
We found that different FileScanTask objects in the taskGroup were pointing to the same data file, causing duplicates. I used putIfAbsent to ensure a single entry per file and renamed the maps to distinctDataFiles. However after switching to collect FileScanTask objects as a set, putIfAbsent is no longer needed. @RussellSpitzer , can you suggest any alternate approach to fix this?

Reply:
Yes, filescan tasks are not guaranteed to have non-overlapping groups of files. That's why i suggested below just collecting them all up in a set before collecting stats. Since all data files are "distinct" by definition we don't really need it in a variable name.

Reply:
Updated to collect data files into a set and renamed the variables, Thanks!